content/en/docs/installing-on-kubernetes/on-prem-kubernetes/install-ks-on-linux-airgapped.md

```diff
@@ -1,10 +1,409 @@
+
+{{</ notice >}}
+
+### Configure Registry
+
+1. Add an entry to `/etc/hosts` to map the hostname (i.e. the registry domain name; in this case, it is `dockerhub.kubekey.local`) to the private IP address of your machine as below.
+
+   ```bash
+   # docker registry
+   192.168.0.2 dockerhub.kubekey.local
+   ```
+
+2. Execute the following commands to copy the certificate to a specified directory and make Docker trust it.
+
+   ```bash
+   mkdir -p  /etc/docker/certs.d/dockerhub.kubekey.local
+   ```
+
+   ```bash
+   cp certs/domain.crt  /etc/docker/certs.d/dockerhub.kubesphere.local/ca.crt
+   ```
+
+   {{< notice note >}}
+
+   The path of the certificate is related to the domain name. When you copy the path, use your actual domain name if it is different from the one set above.
+
+   {{</ notice >}} 
+
+3. To verify whether the private registry is effective, you can copy an image to your local machine first, and use `docker push` and `docker pull` to test it.
+
+## Step 2: Prepare Installation Images
+
+As you install KubeSphere in an air-gapped environment, you need to prepare an image package containing all the necessary images in advance.
+
+1. Download the image list file `images-list.txt` from a machine that has access to the Internet through the following command:
+
+   ```bash
+   curl -L -O https://github.com/kubesphere/ks-installer/releases/download/v3.0.0/images-list.txt
+   ```
+
+   {{< notice note >}}
+
+   This file lists images under `##+modulename` based on different modules. You can add your own images to this file following the same rule. To view the complete file, see [Appendix](../install-ks-on-linux-airgapped/#image-list-of-kubesphere-v300).
+
+   {{</ notice >}} 
+
+2. Download `images-manager.sh`. 
+
+   ```bash
+   curl -L -O https://github.com/kubesphere/ks-installer/releases/download/v3.0.0/offline-installation-tool.sh
+   ```
+
+3. Make the `.sh` file executable.
+
+   ```bash
+   chmod +x offline-installation-tool.sh
+   ```
+
+4. You can execute the command `./offline-installation-tool.sh -h` to see how to use the script:
+
+   ```bash
+   root@master:/home/ubuntu# ./offline-installation-tool.sh -h
+   Usage:
+   
+     ./offline-installation-tool.sh [-l IMAGES-LIST] [-d IMAGES-DIR] [-r PRIVATE-REGISTRY] [-v KUBERNETES-VERSION ]
+   
+   Description:
+     -b                     : save kubernetes' binaries.
+     -d IMAGES-DIR          : the dir of files (tar.gz) which generated by `docker save`. default: /home/ubuntu/kubesphere-images
+     -l IMAGES-LIST         : text file with list of images.
+     -r PRIVATE-REGISTRY    : target private registry:port.
+     -s                     : save model will be applied. Pull the images in the IMAGES-LIST and save images as a tar.gz file.
+     -v KUBERNETES-VERSION  : download kubernetes' binaries. default: v1.17.9
+     -h                     : usage message
+   ```
+
+5. Pull images in the file `images-list.txt`.
```

Comment:
This should be changed to offline-installation-tool.sh.

Reply:
ok , I will change this.

Reply:
OK, I will try this.

Reply:
Changed.